core/src/main/scala/cats/Inject.scala

```diff
@@ -0,0 +1,54 @@
+package cats
+
+/**
+ * Inject is a type class to inject a value of type `A` into a
+ * coproduct of types `B` when `A :≺: B`. The coproduct `B` is
+ * modelled with `Either`.
```

Comment:
Agreed with @kailuowang; I don't think this is particularly informative as to what `Inject` does, especially because we're likely going to provide these other instances in the future from cats.  I'd pick something like:

Inject is a type class providing an injection `f` from type `A` into another type `B`.
An injection is a function which does not destroy any information, and thus for every `b: B` there is at most one `a: A` such that `f(a) = b`.

Because of this all injections admit partial inverses, functions `B => Option[A]`, which pair a value `b: B` with the single value `a: A` that `f` sends to `b` if it exists.